// fakergem/lib/faker/ancient.js
'use strict';

Object.defineProperty(exports, "__esModule", {
  value: true
});

var _createClass = function () { function defineProperties(target, props) { for (var i = 0; i < props.length; i++) { var descriptor = props[i]; descriptor.enumerable = descriptor.enumerable || false; descriptor.configurable = true; if ("value" in descriptor) descriptor.writable = true; Object.defineProperty(target, descriptor.key, descriptor); } } return function (Constructor, protoProps, staticProps) { if (protoProps) defineProperties(Constructor.prototype, protoProps); if (staticProps) defineProperties(Constructor, staticProps); return Constructor; }; }();

function _classCallCheck(instance, Constructor) { if (!(instance instanceof Constructor)) { throw new TypeError("Cannot call a class as a function"); } }

var data = require('../../data/ancient.json');

var Ancient = function () {
  function Ancient(faker) {
    _classCallCheck(this, Ancient);

    this.faker = faker;
  }

  _createClass(Ancient, [{
    key: 'god',
    value: function god() {
      return this.faker.Random.element(data['gods']);
    }
  }, {
    key: 'primordial',
    value: function primordial() {
      return this.faker.Random.element(data['primordials']);
    }
  }, {
    key: 'titan',
    value: function titan() {
      return this.faker.Random.element(data['titans']);
    }
  }, {
    key: 'hero',
    value: function hero() {
      return this.faker.Random.element(data['heroes']);
    }
  }]);

  return Ancient;
}();

exports.default = Ancient;
