// fakergem/lib/faker/beer.js
'use strict';

Object.defineProperty(exports, "__esModule", {
  value: true
});

var _createClass = function () { function defineProperties(target, props) { for (var i = 0; i < props.length; i++) { var descriptor = props[i]; descriptor.enumerable = descriptor.enumerable || false; descriptor.configurable = true; if ("value" in descriptor) descriptor.writable = true; Object.defineProperty(target, descriptor.key, descriptor); } } return function (Constructor, protoProps, staticProps) { if (protoProps) defineProperties(Constructor.prototype, protoProps); if (staticProps) defineProperties(Constructor, staticProps); return Constructor; }; }();

function _classCallCheck(instance, Constructor) { if (!(instance instanceof Constructor)) { throw new TypeError("Cannot call a class as a function"); } }

var data = require('../../data/beer.json');

var Beer = function () {
  function Beer(faker) {
    _classCallCheck(this, Beer);

    this.faker = faker;
  }

  _createClass(Beer, [{
    key: 'name',
    value: function name() {
      return this.faker.Random.element(data['names']);
    }
  }, {
    key: 'style',
    value: function style() {
      return this.faker.Random.element(data['styles']);
    }
  }, {
    key: 'hop',
    value: function hop() {
      return this.faker.Random.element(data['hops']);
    }
  }, {
    key: 'yeast',
    value: function yeast() {
      return this.faker.Random.element(data['yeasts']);
    }
  }, {
    key: 'malt',
    value: function malt() {
      return this.faker.Random.element(data['malts']);
    }
  }, {
    key: 'ibu',
    value: function ibu() {
      return this.faker.Number.between(10, 99) + ' IBU';
    }
  }, {
    key: 'alcohol',
    value: function alcohol() {
      return this.faker.Number.betweenF(2.0, 10.0).toFixed(1) + '%';
    }
  }, {
    key: 'blg',
    value: function blg() {
      return this.faker.Number.betweenF(2.0, 10.0).toFixed(1) + '\xB0Blg';
    }
  }]);

  return Beer;
}();

exports.default = Beer;
