// fakergem/lib/faker/book.js
'use strict';

Object.defineProperty(exports, "__esModule", {
  value: true
});

var _createClass = function () { function defineProperties(target, props) { for (var i = 0; i < props.length; i++) { var descriptor = props[i]; descriptor.enumerable = descriptor.enumerable || false; descriptor.configurable = true; if ("value" in descriptor) descriptor.writable = true; Object.defineProperty(target, descriptor.key, descriptor); } } return function (Constructor, protoProps, staticProps) { if (protoProps) defineProperties(Constructor.prototype, protoProps); if (staticProps) defineProperties(Constructor, staticProps); return Constructor; }; }();

function _classCallCheck(instance, Constructor) { if (!(instance instanceof Constructor)) { throw new TypeError("Cannot call a class as a function"); } }

var data = require('../../data/book.json');

var Book = function () {
  function Book(faker) {
    _classCallCheck(this, Book);

    this.faker = faker;
  }

  _createClass(Book, [{
    key: 'title',
    value: function title() {
      return this.faker.Random.element(data['titles']);
    }
  }, {
    key: 'author',
    value: function author() {
      return [this.faker.Name.firstName(), this.faker.Name.lastName()].join(' ');
    }
  }, {
    key: 'publisher',
    value: function publisher() {
      return this.faker.Random.element(data['publishers']);
    }
  }, {
    key: 'genre',
    value: function genre() {
      return this.faker.Random.element(data['genres']);
    }
  }]);

  return Book;
}();

exports.default = Book;
