// fakergem/lib/faker/business.js
'use strict';

Object.defineProperty(exports, "__esModule", {
  value: true
});

var _createClass = function () { function defineProperties(target, props) { for (var i = 0; i < props.length; i++) { var descriptor = props[i]; descriptor.enumerable = descriptor.enumerable || false; descriptor.configurable = true; if ("value" in descriptor) descriptor.writable = true; Object.defineProperty(target, descriptor.key, descriptor); } } return function (Constructor, protoProps, staticProps) { if (protoProps) defineProperties(Constructor.prototype, protoProps); if (staticProps) defineProperties(Constructor, staticProps); return Constructor; }; }();

function _classCallCheck(instance, Constructor) { if (!(instance instanceof Constructor)) { throw new TypeError("Cannot call a class as a function"); } }

var data = require('../../data/business.json');

function daysFromNow(n) {
  var d = new Date();
  d.setDate(d.getDate() + n);
  d.setHours(0, 0, 0, 0);
  return d;
}

var Business = function () {
  function Business(faker) {
    _classCallCheck(this, Business);

    this.faker = faker;
  }

  _createClass(Business, [{
    key: 'creditCardNumber',
    value: function creditCardNumber() {
      return this.faker.Random.element(data['creditCardNumbers']);
    }
  }, {
    key: 'creditCardExpiryDate',
    value: function creditCardExpiryDate() {
      return daysFromNow(365 * this.faker.Number.between(1, 5));
    }
  }, {
    key: 'creditCardType',
    value: function creditCardType() {
      return this.faker.Random.element(data['creditCardTypes']);
    }
  }]);

  return Business;
}();

exports.default = Business;
