// fakergem/lib/faker/game-of-thrones.js
'use strict';

Object.defineProperty(exports, "__esModule", {
  value: true
});

var _createClass = function () { function defineProperties(target, props) { for (var i = 0; i < props.length; i++) { var descriptor = props[i]; descriptor.enumerable = descriptor.enumerable || false; descriptor.configurable = true; if ("value" in descriptor) descriptor.writable = true; Object.defineProperty(target, descriptor.key, descriptor); } } return function (Constructor, protoProps, staticProps) { if (protoProps) defineProperties(Constructor.prototype, protoProps); if (staticProps) defineProperties(Constructor, staticProps); return Constructor; }; }();

function _classCallCheck(instance, Constructor) { if (!(instance instanceof Constructor)) { throw new TypeError("Cannot call a class as a function"); } }

var data = require('../../data/game-of-thrones.json');

var GameOfThrones = function () {
  function GameOfThrones(faker) {
    _classCallCheck(this, GameOfThrones);

    this.faker = faker;
  }

  _createClass(GameOfThrones, [{
    key: 'character',
    value: function character() {
      return this.faker.Random.element(data['characters']);
    }
  }, {
    key: 'house',
    value: function house() {
      return this.faker.Random.element(data['houses']);
    }
  }, {
    key: 'city',
    value: function city() {
      return this.faker.Random.element(data['cities']);
    }
  }, {
    key: 'dragon',
    value: function dragon() {
      return this.faker.Random.element(data['dragons']);
    }
  }]);

  return GameOfThrones;
}();

exports.default = GameOfThrones;
