// fakergem/lib/faker/harry-potter.js
'use strict';

Object.defineProperty(exports, "__esModule", {
  value: true
});

var _createClass = function () { function defineProperties(target, props) { for (var i = 0; i < props.length; i++) { var descriptor = props[i]; descriptor.enumerable = descriptor.enumerable || false; descriptor.configurable = true; if ("value" in descriptor) descriptor.writable = true; Object.defineProperty(target, descriptor.key, descriptor); } } return function (Constructor, protoProps, staticProps) { if (protoProps) defineProperties(Constructor.prototype, protoProps); if (staticProps) defineProperties(Constructor, staticProps); return Constructor; }; }();

function _classCallCheck(instance, Constructor) { if (!(instance instanceof Constructor)) { throw new TypeError("Cannot call a class as a function"); } }

var data = require('../../data/harry-potter.json');

var HarryPotter = function () {
  function HarryPotter(faker) {
    _classCallCheck(this, HarryPotter);

    this.faker = faker;
  }

  _createClass(HarryPotter, [{
    key: 'character',
    value: function character() {
      return this.faker.Random.element(data['characters']);
    }
  }, {
    key: 'location',
    value: function location() {
      return this.faker.Random.element(data['locations']);
    }
  }, {
    key: 'quote',
    value: function quote() {
      return this.faker.Random.element(data['quotes']);
    }
  }, {
    key: 'book',
    value: function book() {
      return this.faker.Random.element(data['books']);
    }
  }]);

  return HarryPotter;
}();

exports.default = HarryPotter;
