// fakergem/lib/faker/lorem-pixel.js
'use strict';

Object.defineProperty(exports, "__esModule", {
  value: true
});

var _createClass = function () { function defineProperties(target, props) { for (var i = 0; i < props.length; i++) { var descriptor = props[i]; descriptor.enumerable = descriptor.enumerable || false; descriptor.configurable = true; if ("value" in descriptor) descriptor.writable = true; Object.defineProperty(target, descriptor.key, descriptor); } } return function (Constructor, protoProps, staticProps) { if (protoProps) defineProperties(Constructor.prototype, protoProps); if (staticProps) defineProperties(Constructor, staticProps); return Constructor; }; }();

function _classCallCheck(instance, Constructor) { if (!(instance instanceof Constructor)) { throw new TypeError("Cannot call a class as a function"); } }

function _toConsumableArray(arr) { if (Array.isArray(arr)) { for (var i = 0, arr2 = Array(arr.length); i < arr.length; i++) { arr2[i] = arr[i]; } return arr2; } else { return Array.from(arr); } }

var SUPPORTED_CATEGORIES = ['abstract', 'animals', 'business', 'cats', 'city', 'food', 'nightlife', 'fashion', 'people', 'nature', 'sports', 'technics', 'transport'];

function isValidSize(size) {
  return (/^\d+x\d+$/.test(size)
  );
}

function isCategorySupported(category) {
  return !category || SUPPORTED_CATEGORIES.indexOf(category) >= 0;
}

function isNumberValid(number) {
  return number === null || [].concat(_toConsumableArray(Array(10).keys())).map(function (i) {
    return i + 1;
  }).indexOf(parseInt(number)) >= 0;
}

var LoremPixel = function () {
  function LoremPixel(faker) {
    _classCallCheck(this, LoremPixel);

    this.faker = faker;
  }

  _createClass(LoremPixel, [{
    key: 'image',
    value: function image() {
      var size = arguments.length > 0 && arguments[0] !== undefined ? arguments[0] : '300x300';
      var isGray = arguments.length > 1 && arguments[1] !== undefined ? arguments[1] : false;
      var category = arguments.length > 2 && arguments[2] !== undefined ? arguments[2] : null;
      var number = arguments.length > 3 && arguments[3] !== undefined ? arguments[3] : null;
      var text = arguments.length > 4 && arguments[4] !== undefined ? arguments[4] : null;

      if (!isValidSize(size)) {
        throw new Error('Size should be specified in format 300x300');
      }

      if (!isCategorySupported(category)) {
        throw new Error('Supported categories are ' + SUPPORTED_CATEGORIES.join(', '));
      }

      if (!isNumberValid(number)) {
        throw new Error('Number must be between 1 and 10');
      }

      if (number && !category) {
        throw new Error('Category required when number is passed');
      }

      if (text && !category && !number) {
        throw new Error('Category and number must be passed when text is passed');
      }

      return ['http://lorempixel.com', isGray ? 'g' : null, size.replace(/x/, '/'), [category, number, text].filter(function (x) {
        return x;
      }).join('/')].filter(function (x) {
        return x;
      }).join('/');
    }
  }]);

  return LoremPixel;
}();

exports.default = LoremPixel;
