// fakergem/lib/faker/lorem.js
'use strict';

Object.defineProperty(exports, "__esModule", {
  value: true
});

var _createClass = function () { function defineProperties(target, props) { for (var i = 0; i < props.length; i++) { var descriptor = props[i]; descriptor.enumerable = descriptor.enumerable || false; descriptor.configurable = true; if ("value" in descriptor) descriptor.writable = true; Object.defineProperty(target, descriptor.key, descriptor); } } return function (Constructor, protoProps, staticProps) { if (protoProps) defineProperties(Constructor.prototype, protoProps); if (staticProps) defineProperties(Constructor, staticProps); return Constructor; }; }();

function _classCallCheck(instance, Constructor) { if (!(instance instanceof Constructor)) { throw new TypeError("Cannot call a class as a function"); } }

function _toConsumableArray(arr) { if (Array.isArray(arr)) { for (var i = 0, arr2 = Array(arr.length); i < arr.length; i++) { arr2[i] = arr[i]; } return arr2; } else { return Array.from(arr); } }

var data = require('../../data/lorem.json');

// 0-9, a-z
var CHARACTERS = [].concat(_toConsumableArray(Array(10).keys())).concat([].concat(_toConsumableArray(Array(26).keys())).map(function (i) {
  return String.fromCharCode(97 + i);
}));

function resolveNumber(n) {
  return Math.max(parseInt(n), 0);
}

function capitalize(text) {
  return text.replace(/\b\w/, function (x) {
    return x.toUpperCase();
  });
}

var Lorem = function () {
  function Lorem(faker) {
    _classCallCheck(this, Lorem);

    this.faker = faker;
  }

  _createClass(Lorem, [{
    key: 'word',
    value: function word() {
      return this.faker.Random.element(data['words']);
    }
  }, {
    key: 'words',
    value: function words() {
      var num = arguments.length > 0 && arguments[0] !== undefined ? arguments[0] : 3;
      var supplemental = arguments.length > 1 && arguments[1] !== undefined ? arguments[1] : false;

      var wordList = supplemental ? [].concat(_toConsumableArray(data['words']), [data['supplemental']]) : data['words'];
      var words = this.faker.Random.assortment(wordList, resolveNumber(num));
      return words;
    }
  }, {
    key: 'character',
    value: function character() {
      return this.faker.Random.element(CHARACTERS);
    }
  }, {
    key: 'characters',
    value: function characters() {
      var _this = this;

      var charCount = arguments.length > 0 && arguments[0] !== undefined ? arguments[0] : 255;

      return [].concat(_toConsumableArray(Array(resolveNumber(charCount)).keys())).map(function (_) {
        return _this.character();
      }).join('');
    }
  }, {
    key: 'sentence',
    value: function sentence() {
      var wordCount = arguments.length > 0 && arguments[0] !== undefined ? arguments[0] : 4;
      var supplemental = arguments.length > 1 && arguments[1] !== undefined ? arguments[1] : false;
      var randomWordsToAdd = arguments.length > 2 && arguments[2] !== undefined ? arguments[2] : 6;

      var text = capitalize(this.words(resolveNumber(wordCount) + this.faker.Number.between(0, randomWordsToAdd), supplemental).join(' '));
      return text.length > 0 ? text + '.' : text;
    }
  }, {
    key: 'sentences',
    value: function sentences() {
      var _this2 = this;

      var sentenceCount = arguments.length > 0 && arguments[0] !== undefined ? arguments[0] : 3;
      var supplemental = arguments.length > 1 && arguments[1] !== undefined ? arguments[1] : false;

      return [].concat(_toConsumableArray(Array(resolveNumber(sentenceCount)).keys())).map(function (_) {
        return _this2.sentence(3, supplemental);
      });
    }
  }, {
    key: 'paragraph',
    value: function paragraph() {
      var sentenceCount = arguments.length > 0 && arguments[0] !== undefined ? arguments[0] : 3;
      var supplemental = arguments.length > 1 && arguments[1] !== undefined ? arguments[1] : false;
      var randomSentencesToAdd = arguments.length > 2 && arguments[2] !== undefined ? arguments[2] : 3;

      return this.sentences(resolveNumber(sentenceCount) + this.faker.Number.between(0, randomSentencesToAdd), supplemental).join(' ');
    }
  }, {
    key: 'paragraphs',
    value: function paragraphs() {
      var _this3 = this;

      var paragraphCount = arguments.length > 0 && arguments[0] !== undefined ? arguments[0] : 3;
      var supplemental = arguments.length > 1 && arguments[1] !== undefined ? arguments[1] : false;

      return [].concat(_toConsumableArray(Array(resolveNumber(paragraphCount)).keys())).map(function (_) {
        return _this3.paragraph(3, supplemental);
      });
    }
  }]);

  return Lorem;
}();

exports.default = Lorem;
