// fakergem/lib/faker/lorem-flickr.js
'use strict';

Object.defineProperty(exports, "__esModule", {
  value: true
});

var _createClass = function () { function defineProperties(target, props) { for (var i = 0; i < props.length; i++) { var descriptor = props[i]; descriptor.enumerable = descriptor.enumerable || false; descriptor.configurable = true; if ("value" in descriptor) descriptor.writable = true; Object.defineProperty(target, descriptor.key, descriptor); } } return function (Constructor, protoProps, staticProps) { if (protoProps) defineProperties(Constructor.prototype, protoProps); if (staticProps) defineProperties(Constructor, staticProps); return Constructor; }; }();

function _classCallCheck(instance, Constructor) { if (!(instance instanceof Constructor)) { throw new TypeError("Cannot call a class as a function"); } }

var SUPPORTED_COLORIZATIONS = ['red', 'green', 'blue'];

function buildUrl(size, format, searchTerms, matchAll) {
  if (!isValidSize(size)) {
    throw new Error('Size should be specified in format 300x300');
  }

  var url = 'http://loremflickr.com';
  if (format) {
    url += '/' + format;
  }
  url += '/' + size.replace('x', '/');
  if ((searchTerms || []).length > 0) {
    url += '/' + searchTerms.join(',');
  }
  if (matchAll) {
    url += '/all';
  }
  return url;
}

function hasSearchTerms(searchTerms) {
  return (searchTerms || []).length > 0;
}

function isColorizationSupported(color) {
  return SUPPORTED_COLORIZATIONS.indexOf(color) >= 0;
}

function isValidSize(size) {
  return (/^\d+x\d+$/.test(size)
  );
}

var LoremFlickr = function () {
  function LoremFlickr(faker) {
    _classCallCheck(this, LoremFlickr);

    this.faker = faker;
  }

  _createClass(LoremFlickr, [{
    key: 'image',
    value: function image() {
      var size = arguments.length > 0 && arguments[0] !== undefined ? arguments[0] : '300x300';
      var searchTerms = arguments.length > 1 && arguments[1] !== undefined ? arguments[1] : [];
      var matchAll = arguments.length > 2 && arguments[2] !== undefined ? arguments[2] : false;

      return buildUrl(size, null, searchTerms, matchAll);
    }
  }, {
    key: 'grayscaleImage',
    value: function grayscaleImage() {
      var size = arguments.length > 0 && arguments[0] !== undefined ? arguments[0] : '300x300';
      var searchTerms = arguments.length > 1 && arguments[1] !== undefined ? arguments[1] : ['all'];
      var matchAll = arguments.length > 2 && arguments[2] !== undefined ? arguments[2] : false;

      if (!hasSearchTerms(searchTerms)) {
        throw new Error('Search terms must be specified for grayscale images');
      }

      return buildUrl(size, 'g', searchTerms, matchAll);
    }
  }, {
    key: 'pixelatedImage',
    value: function pixelatedImage() {
      var size = arguments.length > 0 && arguments[0] !== undefined ? arguments[0] : '300x300';
      var searchTerms = arguments.length > 1 && arguments[1] !== undefined ? arguments[1] : ['all'];
      var matchAll = arguments.length > 2 && arguments[2] !== undefined ? arguments[2] : false;

      if (!hasSearchTerms(searchTerms)) {
        throw new Error('Search terms must be specified for pixelated images');
      }

      return buildUrl(size, 'p', searchTerms, matchAll);
    }
  }, {
    key: 'colorizedImage',
    value: function colorizedImage() {
      var size = arguments.length > 0 && arguments[0] !== undefined ? arguments[0] : '300x300';
      var color = arguments.length > 1 && arguments[1] !== undefined ? arguments[1] : 'red';
      var searchTerms = arguments.length > 2 && arguments[2] !== undefined ? arguments[2] : ['all'];
      var matchAll = arguments.length > 3 && arguments[3] !== undefined ? arguments[3] : false;

      if (!hasSearchTerms(searchTerms)) {
        throw new Error('Search terms must be specified for colorized images');
      }
      if (!isColorizationSupported(color)) {
        throw new Error('Supported colorizations are ' + SUPPORTED_COLORIZATIONS.join(', '));
      }

      return buildUrl(size, color, searchTerms, matchAll);
    }
  }]);

  return LoremFlickr;
}();

exports.default = LoremFlickr;
