// fakergem/lib/faker/placeholdit.js
'use strict';

Object.defineProperty(exports, "__esModule", {
  value: true
});

var _createClass = function () { function defineProperties(target, props) { for (var i = 0; i < props.length; i++) { var descriptor = props[i]; descriptor.enumerable = descriptor.enumerable || false; descriptor.configurable = true; if ("value" in descriptor) descriptor.writable = true; Object.defineProperty(target, descriptor.key, descriptor); } } return function (Constructor, protoProps, staticProps) { if (protoProps) defineProperties(Constructor.prototype, protoProps); if (staticProps) defineProperties(Constructor, staticProps); return Constructor; }; }();

function _classCallCheck(instance, Constructor) { if (!(instance instanceof Constructor)) { throw new TypeError("Cannot call a class as a function"); } }

var SUPPORTED_FORMATS = ['png', 'jpg', 'gif', 'jpeg'];

function isValidSize(size) {
  return (/^\d+x\d+$/.test(size)
  );
}

function isSupportedFormat(format) {
  return SUPPORTED_FORMATS.indexOf(format) >= 0;
}

function isValidColorValue(color) {
  if (color) {
    return new RegExp(/(?:^[0-9a-f]{3}$)|(?:^[0-9a-f]{6}$)/.source, 'i').test(color);
  }
  return true;
}

var Placeholdit = function () {
  function Placeholdit(faker) {
    _classCallCheck(this, Placeholdit);

    this.faker = faker;
  }

  _createClass(Placeholdit, [{
    key: 'image',
    value: function image() {
      var size = arguments.length > 0 && arguments[0] !== undefined ? arguments[0] : '300x300';
      var format = arguments.length > 1 && arguments[1] !== undefined ? arguments[1] : 'png';
      var backgroundColor = arguments.length > 2 && arguments[2] !== undefined ? arguments[2] : null;
      var textColor = arguments.length > 3 && arguments[3] !== undefined ? arguments[3] : null;
      var text = arguments.length > 4 && arguments[4] !== undefined ? arguments[4] : null;

      if (!isValidSize(size)) {
        throw new Error('Size should be specified in format 300x300');
      }
      if (!isSupportedFormat(format)) {
        throw new Error('Supported formats are ' + SUPPORTED_FORMATS.join(', '));
      }
      if (!isValidColorValue(backgroundColor)) {
        throw new Error("backgroundColor must be a hex value without '#'");
      }
      if (!isValidColorValue(textColor)) {
        throw new Error("textColor must be a hex value without '#'");
      }

      var url = 'https://placehold.it/' + size + '.' + format;
      if (backgroundColor) {
        url += '/' + backgroundColor;
      }
      if (textColor) {
        url += '/' + textColor;
      }
      if (text) {
        url += '?text=' + text;
      }
      return url;
    }
  }]);

  return Placeholdit;
}();

exports.default = Placeholdit;
