// fakergem/lib/faker/superhero.js
'use strict';

Object.defineProperty(exports, "__esModule", {
  value: true
});

var _createClass = function () { function defineProperties(target, props) { for (var i = 0; i < props.length; i++) { var descriptor = props[i]; descriptor.enumerable = descriptor.enumerable || false; descriptor.configurable = true; if ("value" in descriptor) descriptor.writable = true; Object.defineProperty(target, descriptor.key, descriptor); } } return function (Constructor, protoProps, staticProps) { if (protoProps) defineProperties(Constructor.prototype, protoProps); if (staticProps) defineProperties(Constructor, staticProps); return Constructor; }; }();

function _classCallCheck(instance, Constructor) { if (!(instance instanceof Constructor)) { throw new TypeError("Cannot call a class as a function"); } }

var data = require('../../data/superhero.json');

var Superhero = function () {
  function Superhero(faker) {
    _classCallCheck(this, Superhero);

    this.faker = faker;
  }

  _createClass(Superhero, [{
    key: 'name',
    value: function name() {
      return this.faker.Random.element(data['names']).replace('{prefix}', this.faker.Random.element(data['prefixes'])).replace('{suffix}', this.faker.Random.element(data['suffixes'])).replace('{descriptor}', this.faker.Random.element(data['descriptors']));
    }
  }, {
    key: 'power',
    value: function power() {
      return this.faker.Random.element(data['powers']);
    }
  }, {
    key: 'prefix',
    value: function prefix() {
      return this.faker.Random.element(data['prefixes']);
    }
  }, {
    key: 'suffix',
    value: function suffix() {
      return this.faker.Random.element(data['suffixes']);
    }
  }, {
    key: 'descriptor',
    value: function descriptor() {
      return this.faker.Random.element(data['descriptors']);
    }
  }]);

  return Superhero;
}();

exports.default = Superhero;
