// fakergem/lib/faker/team.js
'use strict';

Object.defineProperty(exports, "__esModule", {
  value: true
});

var _createClass = function () { function defineProperties(target, props) { for (var i = 0; i < props.length; i++) { var descriptor = props[i]; descriptor.enumerable = descriptor.enumerable || false; descriptor.configurable = true; if ("value" in descriptor) descriptor.writable = true; Object.defineProperty(target, descriptor.key, descriptor); } } return function (Constructor, protoProps, staticProps) { if (protoProps) defineProperties(Constructor.prototype, protoProps); if (staticProps) defineProperties(Constructor, staticProps); return Constructor; }; }();

function _classCallCheck(instance, Constructor) { if (!(instance instanceof Constructor)) { throw new TypeError("Cannot call a class as a function"); } }

var data = require('../../data/team.json');

var Team = function () {
  function Team(faker) {
    _classCallCheck(this, Team);

    this.faker = faker;
  }

  _createClass(Team, [{
    key: 'creature',
    value: function creature() {
      return this.faker.Random.element(data['creatures']);
    }
  }, {
    key: 'name',
    value: function name() {
      return this.faker.Random.element(data['names']);
    }
  }, {
    key: 'state',
    value: function state() {
      return this.faker.Address.state();
    }
  }, {
    key: 'sport',
    value: function sport() {
      return this.faker.Random.element(data['sports']);
    }
  }]);

  return Team;
}();

exports.default = Team;
