// fakergem/lib/faker.js
'use strict';

var _address = require('./faker/address');

var _address2 = _interopRequireDefault(_address);

var _ancient = require('./faker/ancient');

var _ancient2 = _interopRequireDefault(_ancient);

var _app = require('./faker/app');

var _app2 = _interopRequireDefault(_app);

var _avatar = require('./faker/avatar');

var _avatar2 = _interopRequireDefault(_avatar);

var _bank = require('./faker/bank');

var _bank2 = _interopRequireDefault(_bank);

var _beer = require('./faker/beer');

var _beer2 = _interopRequireDefault(_beer);

var _bitcoin = require('./faker/bitcoin');

var _bitcoin2 = _interopRequireDefault(_bitcoin);

var _book = require('./faker/book');

var _book2 = _interopRequireDefault(_book);

var _boolean = require('./faker/boolean');

var _boolean2 = _interopRequireDefault(_boolean);

var _business = require('./faker/business');

var _business2 = _interopRequireDefault(_business);

var _cat = require('./faker/cat');

var _cat2 = _interopRequireDefault(_cat);

var _chuckNorris = require('./faker/chuck-norris');

var _chuckNorris2 = _interopRequireDefault(_chuckNorris);

var _coffee = require('./faker/coffee');

var _coffee2 = _interopRequireDefault(_coffee);

var _color = require('./faker/color');

var _color2 = _interopRequireDefault(_color);

var _commerce = require('./faker/commerce');

var _commerce2 = _interopRequireDefault(_commerce);

var _company = require('./faker/company');

var _company2 = _interopRequireDefault(_company);

var _compass = require('./faker/compass');

var _compass2 = _interopRequireDefault(_compass);

var _crypto = require('./faker/crypto');

var _crypto2 = _interopRequireDefault(_crypto);

var _date = require('./faker/date');

var _date2 = _interopRequireDefault(_date);

var _demographic = require('./faker/demographic');

var _demographic2 = _interopRequireDefault(_demographic);

var _dragonBall = require('./faker/dragon-ball');

var _dragonBall2 = _interopRequireDefault(_dragonBall);

var _educator = require('./faker/educator');

var _educator2 = _interopRequireDefault(_educator);

var _fake = require('./faker/fake');

var _fake2 = _interopRequireDefault(_fake);

var _file = require('./faker/file');

var _file2 = _interopRequireDefault(_file);

var _fillmurray = require('./faker/fillmurray');

var _fillmurray2 = _interopRequireDefault(_fillmurray);

var _food = require('./faker/food');

var _food2 = _interopRequireDefault(_food);

var _friends = require('./faker/friends');

var _friends2 = _interopRequireDefault(_friends);

var _gameOfThrones = require('./faker/game-of-thrones');

var _gameOfThrones2 = _interopRequireDefault(_gameOfThrones);

var _hacker = require('./faker/hacker');

var _hacker2 = _interopRequireDefault(_hacker);

var _harryPotter = require('./faker/harry-potter');

var _harryPotter2 = _interopRequireDefault(_harryPotter);

var _hipster = require('./faker/hipster');

var _hipster2 = _interopRequireDefault(_hipster);

var _internet = require('./faker/internet');

var _internet2 = _interopRequireDefault(_internet);

var _lordOfTheRings = require('./faker/lord-of-the-rings');

var _lordOfTheRings2 = _interopRequireDefault(_lordOfTheRings);

var _loremFlickr = require('./faker/lorem-flickr');

var _loremFlickr2 = _interopRequireDefault(_loremFlickr);

var _loremPixel = require('./faker/lorem-pixel');

var _loremPixel2 = _interopRequireDefault(_loremPixel);

var _lorem = require('./faker/lorem');

var _lorem2 = _interopRequireDefault(_lorem);

var _matz = require('./faker/matz');

var _matz2 = _interopRequireDefault(_matz);

var _music = require('./faker/music');

var _music2 = _interopRequireDefault(_music);

var _name = require('./faker/name');

var _name2 = _interopRequireDefault(_name);

var _number = require('./faker/number');

var _number2 = _interopRequireDefault(_number);

var _phoneNumber = require('./faker/phone-number');

var _phoneNumber2 = _interopRequireDefault(_phoneNumber);

var _placeholdit = require('./faker/placeholdit');

var _placeholdit2 = _interopRequireDefault(_placeholdit);

var _pokemon = require('./faker/pokemon');

var _pokemon2 = _interopRequireDefault(_pokemon);

var _random = require('./faker/random');

var _random2 = _interopRequireDefault(_random);

var _rickAndMorty = require('./faker/rick-and-morty');

var _rickAndMorty2 = _interopRequireDefault(_rickAndMorty);

var _rockBand = require('./faker/rock-band');

var _rockBand2 = _interopRequireDefault(_rockBand);

var _ruPaul = require('./faker/ru-paul');

var _ruPaul2 = _interopRequireDefault(_ruPaul);

var _slackEmoji = require('./faker/slack-emoji');

var _slackEmoji2 = _interopRequireDefault(_slackEmoji);

var _space = require('./faker/space');

var _space2 = _interopRequireDefault(_space);

var _starWars = require('./faker/star-wars');

var _starWars2 = _interopRequireDefault(_starWars);

var _superhero = require('./faker/superhero');

var _superhero2 = _interopRequireDefault(_superhero);

var _team = require('./faker/team');

var _team2 = _interopRequireDefault(_team);

var _time = require('./faker/time');

var _time2 = _interopRequireDefault(_time);

var _twinPeaks = require('./faker/twin-peaks');

var _twinPeaks2 = _interopRequireDefault(_twinPeaks);

var _twitter = require('./faker/twitter');

var _twitter2 = _interopRequireDefault(_twitter);

var _university = require('./faker/university');

var _university2 = _interopRequireDefault(_university);

var _vehicle = require('./faker/vehicle');

var _vehicle2 = _interopRequireDefault(_vehicle);

var _zelda = require('./faker/zelda');

var _zelda2 = _interopRequireDefault(_zelda);

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

function _classCallCheck(instance, Constructor) { if (!(instance instanceof Constructor)) { throw new TypeError("Cannot call a class as a function"); } }

var Faker = function Faker() {
  _classCallCheck(this, Faker);

  this.Address = new _address2.default(this);
  this.Ancient = new _ancient2.default(this);
  this.App = new _app2.default(this);
  this.Avatar = new _avatar2.default(this);
  this.Bank = new _bank2.default(this);
  this.Beer = new _beer2.default(this);
  this.Bitcoin = new _bitcoin2.default(this);
  this.Book = new _book2.default(this);
  this.Boolean = new _boolean2.default(this);
  this.Business = new _business2.default(this);
  this.Cat = new _cat2.default(this);
  this.ChuckNorris = new _chuckNorris2.default(this);
  this.Coffee = new _coffee2.default(this);
  this.Color = new _color2.default(this);
  this.Commerce = new _commerce2.default(this);
  this.Company = new _company2.default(this);
  this.Compass = new _compass2.default(this);
  this.Crypto = new _crypto2.default(this);
  this.Date = new _date2.default(this);
  this.Demographic = new _demographic2.default(this);
  this.DragonBall = new _dragonBall2.default(this);
  this.Educator = new _educator2.default(this);
  this.Fake = new _fake2.default(this);
  this.File = new _file2.default(this);
  this.Fillmurray = new _fillmurray2.default(this);
  this.Food = new _food2.default(this);
  this.Friends = new _friends2.default(this);
  this.GameOfThrones = new _gameOfThrones2.default(this);
  this.Hacker = new _hacker2.default(this);
  this.HarryPotter = new _harryPotter2.default(this);
  this.Hipster = new _hipster2.default(this);
  this.Internet = new _internet2.default(this);
  this.LordOfTheRings = new _lordOfTheRings2.default(this);
  this.LoremFlickr = new _loremFlickr2.default(this);
  this.LoremPixel = new _loremPixel2.default(this);
  this.Lorem = new _lorem2.default(this);
  this.Matz = new _matz2.default(this);
  this.Music = new _music2.default(this);
  this.Name = new _name2.default(this);
  this.Number = new _number2.default(this);
  this.PhoneNumber = new _phoneNumber2.default(this);
  this.Placeholdit = new _placeholdit2.default(this);
  this.Pokemon = new _pokemon2.default(this);
  this.Random = new _random2.default(this);
  this.RickAndMorty = new _rickAndMorty2.default(this);
  this.RockBand = new _rockBand2.default(this);
  this.RuPaul = new _ruPaul2.default(this);
  this.SlackEmoji = new _slackEmoji2.default(this);
  this.Space = new _space2.default(this);
  this.StarWars = new _starWars2.default(this);
  this.Superhero = new _superhero2.default(this);
  this.Team = new _team2.default(this);
  this.Time = new _time2.default(this);
  this.TwinPeaks = new _twinPeaks2.default(this);
  this.Twitter = new _twitter2.default(this);
  this.University = new _university2.default(this);
  this.Vehicle = new _vehicle2.default(this);
  this.Zelda = new _zelda2.default(this);
};

module.exports = new Faker();
